### PR TITLE
Update snowflake-configs.md

### PR DESCRIPTION
**Pull request by amychen1776**
## What are you changing in this pull request and why?
Resolving Issue #4869 

## Checklist

- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [ ] For [docs versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#about-versioning), review how to [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content).
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding or removing pages (delete if not applicable):
- [ ] Add/remove page in `website/sidebars.js`
- [ ] Provide a unique filename for new pages
- [ ] Add an entry for deleted pages in `website/vercel.json`
- [ ] Run link testing locally with `npm run build` to update the links that point to deleted pages

### CODE REVIEW

**Review by nataliefiann — APPROVED** [review]
Hiya @amychen1776 

Thanks for flagging this - this was really good. I'll approve this for you.
As a non-blocker suggestion, would it be worthwhile linking to the Model Contracts page and the Copy Grants config section?

Kind Regards

File: website/docs/reference/resource-configs/snowflake-configs.md [hunk 1]
```diff
@@ -574,6 +574,10 @@ As with materialized views on most data platforms, there are limitations associa
 
 Find more information about dynamic table limitations in Snowflake's [docs](https://docs.snowflake.com/en/user-guide/dynamic-tables-tasks-create#dynamic-table-limitations-and-supported-functions).
 
+For dbt limitations, these dbt features are not supported:
+- Model Contracts
```
Comment (amychen1776) [inline website/docs/reference/resource-configs/snowflake-configs.md]:
```suggestion
- [Model Contracts](/collaborate/govern/model-contracts)
```

File: website/docs/reference/resource-configs/snowflake-configs.md [hunk 2]
```diff
@@ -574,6 +574,10 @@ As with materialized views on most data platforms, there are limitations associa
...
+- Copy Grants configration
```
Comment (amychen1776) [inline website/docs/reference/resource-configs/snowflake-configs.md]:
```suggestion
- [Copy Grants configuration](/snowflake-configs#copying-grants)
```

File: website/docs/reference/resource-configs/snowflake-configs.md [hunk 3]
```diff
@@ -574,6 +574,10 @@ As with materialized views on most data platforms, there are limitations associa
 
 Find more information about dynamic table limitations in Snowflake's [docs](https://docs.snowflake.com/en/user-guide/dynamic-tables-tasks-create#dynamic-table-limitations-and-supported-functions).
 
+For dbt limitations, these dbt features are not supported:
+- [Model Contracts](/collaborate/govern/model-contracts)
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/snowflake-configs.md]:
```suggestion
- [Model contracts](/docs/collaborate/govern/model-contracts)
```

File: website/docs/reference/resource-configs/snowflake-configs.md [hunk 4]
```diff
@@ -574,6 +574,10 @@ As with materialized views on most data platforms, there are limitations associa
...
+- [Copy Grants configuration](/snowflake-configs#copying-grants)
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/snowflake-configs.md]:
```suggestion
- [Copy grants configuration](/reference/resource-configs/snowflake-configs#copying-grants)
```

**Review by nataliefiann — APPROVED** [review]
Hiya @amychen1776 

The changes are great! Thanks for that. I've approved this for you and will merge this for you.

Kind Regards
Natalie

**Pull request merged**